### PR TITLE
(BKR-895) pull puppet-specific functionality out of beaker

### PR DESCRIPTION
**Pull request by kevpl**
### Tasks to complete before merging

- [x] create automated testing for beaker-puppet PRs
- [x] release a first beaker-puppet gem
- Fix up copies of these jobs to use beaker-puppet
  - [x] [Beaker Acceptance - Reuse Vpool Hosts](https://jenkins-beaker.delivery.puppetlabs.net/job/qe_beaker_intn-sys_beaker-acceptance-reuse-vpool/) ([copied job](https://jenkins-beaker.delivery.puppetlabs.net/job/qe_beaker_intn-sys_beaker-acceptance-reuse-vpool-debug-beaker-puppet/))
  - [x] [Beaker Acceptance - Puppet](https://jenkins-beaker.delivery.puppetlabs.net/job/qe_beaker_intn-sys_beaker-acceptance-puppet-vpool/) ([copied job](https://jenkins-beaker.delivery.puppetlabs.net/job/qe_beaker_intn-sys_beaker-acceptance-puppet-vpool-puppet-pkg-bkr895/))

### CODE REVIEW

**Comment by kevpl** [comment]
@puppetlabs/beaker, @tvpartytonight, would you please take a look at [beaker-puppet](https://github.com/puppetlabs/beaker-puppet)? I'd like to get a review pass before cutting the first gem to be used for last testing with this code.

Also, note that you can explore testing on that repo by checking out [beaker-puppet#1](https://github.com/puppetlabs/beaker-puppet/pull/1).

File: beaker.gemspec [hunk 1]
```diff
@@ -48,6 +48,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency 'stringify-hash', '~> 0.0'
   s.add_runtime_dependency 'beaker-hiera', '~> 0.0'
   s.add_runtime_dependency 'beaker-hostgenerator'
+  s.add_runtime_dependency 'beaker-puppet', '~> 0.0'
```
Comment (er0ck) [inline beaker.gemspec]: won't this only install 0.x versions, requiring a change later when these gems get to 1.x+?

sometimes i use `~> 0`

File: lib/beaker/dsl/helpers.rb [hunk 2]
```diff
@@ -23,14 +24,12 @@ module DSL
...
     module Helpers
-      include Beaker::DSL::Helpers::FacterHelpers
       include Beaker::DSL::Helpers::HostHelpers
-      include Beaker::DSL::Helpers::PuppetHelpers
       include Beaker::DSL::Helpers::TestHelpers
-      include Beaker::DSL::Helpers::TKHelpers
       include Beaker::DSL::Helpers::WebHelpers
       include Beaker::DSL::Helpers::HoconHelpers
       include Beaker::DSL::Helpers::Hiera
+      include BeakerPuppet::Helpers
```
Comment (er0ck) [inline lib/beaker/dsl/helpers.rb]: different pattern. 
should we use a better module naming scheme for beaker-hiera?
we probably shouldn't care much as hiera as a gem is going away (it's internal to puppet now)

File: lib/beaker/dsl/helpers.rb [hunk 3]
```diff
@@ -23,14 +24,12 @@ module DSL
     #
     #
     module Helpers
-      include Beaker::DSL::Helpers::FacterHelpers
       include Beaker::DSL::Helpers::HostHelpers
-      include Beaker::DSL::Helpers::PuppetHelpers
       include Beaker::DSL::Helpers::TestHelpers
-      include Beaker::DSL::Helpers::TKHelpers
```
Comment (er0ck) [inline lib/beaker/dsl/helpers.rb]: where are the tk-helpers going?  into beaker-puppet? 
or are we just eliminating the auto-include in this vesion for tkhelpers and facter?

Comment (er0ck) [inline lib/beaker/dsl/helpers.rb]: it looks like facter, tk and puppet helper are all going into beaker-puppet (https://github.com/puppetlabs/beaker-puppet/commit/3878c13311bf5ddcda043f5c23cb1354614a997a) does this mean beaker-facter is going away?  
i doubt beaker-facter is all that heavy, that might be a good idea.  what about beaker-hiera?

File: beaker.gemspec (hunk 1 above)
Comment (kevpl) [inline beaker.gemspec]: @er0ck yes. So far, we've started publishing beaker-libraries at the `0.1.0` version, and used general major version rules for eventually moving to `1.y`, so this is what I'm going for. It's not much of a precedent though, so I'm open to different viewpoints if you'd like to suggest a better way.

I assume at this point that moving from beaker including this gem to having it exist completely independently will be the time for the `1.y` cut, but we'll see what happens in between now and then.

File: lib/beaker/dsl/helpers.rb (hunk 3 above)
Comment (kevpl) [inline lib/beaker/dsl/helpers.rb]: @er0ck yeah, beaker-facter should be going away. It (and beaker-hiera) were good learning exercises, but they don't appear to be enough code at this point to justify existing separately from beaker-puppet. The same is true of tk-helpers. Let me know if you think that's wrong, and if so, we can discuss this further.

Comment (er0ck) [inline lib/beaker/dsl/helpers.rb]: i think it's the right approach.  carry on  👍

**Comment by kevpl** [comment]
now that [beaker-puppet 0.1.0](https://rubygems.org/gems/beaker-puppet) is released, we can look at testing this PR.

@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by melissa** [comment]
@kevpl it looks like we're going to need this soon. RE is getting ready to cut artifact pathing over to puppet5 soon in prep for the puppet-agent, puppetserver, and puppetdb 5.0.0 releases. Do you have an eta on when we can expect this PR and https://github.com/puppetlabs/beaker/pull/1408 to land? Those releases are scheduled for the 27th.

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
look at that! A green run! I only had to disable these jobs (internal links):

- [Beaker Acceptance - Puppet](https://jenkins-beaker.delivery.puppetlabs.net/job/qe_beaker_intn-sys_beaker-acceptance-puppet-vpool/)
- [Beaker Acceptance - Reuse Vpool Hosts](https://jenkins-beaker.delivery.puppetlabs.net/job/qe_beaker_intn-sys_beaker-acceptance-reuse-vpool/)

I'll create some TODOs in the top to get these prepared for merging...

File: lib/beaker/dsl/install_utils.rb [hunk 4]
```diff
@@ -1,17 +1,12 @@
...
-end
 require "beaker/dsl/install_utils/pe_defaults"
+require 'beaker-puppet'
 
 module Beaker
   module DSL
     # Collection of installation methods and support
     module InstallUtils
-      include DSL::InstallUtils::PuppetUtils
       include DSL::InstallUtils::PEDefaults
```
Comment (tvpartytonight) [inline lib/beaker/dsl/install_utils.rb]: Not a blocker for anything, but should this be moved to beaker-pe? Is that ticketed anywhere?

**Pull request merged**